### PR TITLE
feature/SOF 7530

### PR DESCRIPTION
**Pull request by VsevolodX**
- wip: rough implementation of heterostack (takes toooooo long)
- update: create layer by layer with simple builder
- update: cleanups

### CODE REVIEW

**Comment by VsevolodX** [comment]
Resulting material now
image

File: other/materials_designer/specific_examples/heterostructure_high_k_metal_gate.ipynb [hunk 1]
```diff
@@ -0,0 +1,430 @@
+{
```
Comment (timurbazhirov) [inline other/materials_designer/specific_examples/heterostructure_high_k_metal_gate.ipynb:1]: <p>Same here - isolate visualization part</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (timurbazhirov) [inline other/materials_designer/specific_examples/heterostructure_high_k_metal_gate.ipynb:1]: <p>We can split the two steps below</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

File: other/materials_designer/specific_examples/heterostructure_high_k_metal_gate_stack.ipynb [hunk 2]
```diff
@@ -0,0 +1,507 @@
+{
```
Comment (timurbazhirov) [inline other/materials_designer/specific_examples/heterostructure_high_k_metal_gate_stack.ipynb:1]: <p>We need to add a link to the original tutorial</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Exabyte-io/api-examples/pull/185/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**